### DISCUSSION TITLE
First Justpy community meeting

### DISCUSSION THREAD

**Discussion by WolfgangFahl** [elimintz/justpy#406] [General]
The first Justpy community meeting will be online

via Zoom [zoom link](https://rwth.zoom.us/j/98027129594?pwd=NVZMVzg5UDFOMzU0VFdteHN0MWsvZz09)
next Friday August 19,

- 4 PM Berlin Time,
- 5 PM London
- Noon New York
- 9 AM Los Angeles

We'll discuss how to proceed with this project based on #373 and #404

**Comment by freemant2000** [comment]
How did the meeting go? Any resolutions?

**Reply by rodja** [reply]
The meeting will start in 50 minutes.

**Reply by WolfgangFahl** [reply]
The [zoom session](https://rwth.zoom.us/j/98027129594?pwd=NVZMVzg5UDFOMzU0VFdteHN0MWsvZz09) is open now.

**Comment by WolfgangFahl** [comment]
**Agenda:**

- Introduction of participants - who are you - why are you here? what do you expect? what can you/will you contribute?
- What do we want to achieve short term/ long term
- How do we want to do it?
- Todos
- Next meeting

**Participants**
see individual introductions below

- Eli
- Falko
- Rodja
- Sandra
- Tim
- Wolfgang


**Short Term / Long Term goals**

1. AgGrid - integration
2. Tailwind version was outdate - version handling for Quasar and Tailwind
3. stability / debugging - sometimes justpy just hangs 
4. test suite / test first approach
5. resources - time and focus to fix issues
6. Hard to find out what the core functionality is how could justpy be more agnostic of libraries
7. Logging infrastructure
8. Process for Changing 
9. Marketing missing?

**Showcases**

....

**How do we want to do it?**
How to work collaboratively - someone should do it ...
Invite participants as collaborators  - make Wolfgang maintainer to allow more participants
Eli forks a copy for him self so that he can continue to work as a collaborator

**When to meet again / how often**

* Do most of the work offline
* Meet online if necessary - adhoc - with some announcement
* Postpone decision for regular meetings

**Marketing**
Where do we want to position justpy?

Bigger organizations happily embrace the paradigm of separation of front-end and backend. As of now the justpy approach looks like a nice solution for people that have specials needs.

**Comment by WolfgangFahl** [comment]
Who: Wolfgang Fahl - RWTH Aachen i5
Why: justpy is a good tool for prototyping see e.g http://wikidata.bitplan.com  http://ceur-ws-browser.bitplan.com/
see also http://jpdemo.bitplan.com
Expectation: make justpy a well maintained platform
What to offer: SW architecture and requirements engineering expertise, continuous integration

**Comment by WolfgangFahl** [comment]
Sandra does not have her own account yet:
Who: Sandra Fahl, UoFA Edmonton, Mining
Why: potential python GUI platform  for dashboards
Expectation: check potential for python 
What to offer: don't know yet - depends ...

**Comment by rodja** [comment]
Who: Rodja Trappe and Falko Schindler, [Zauberzeug GmbH](https://zauberzeug.com)
Why: we use JustPy as a basis for [NiceGUI](https://nicegui.io) which itself is basis for many of our projects like [RoSys](https://rosys.io), a framework for mobile robotics.
Expectation: keep JustPy alive and let it be a great core engine
What to offer: software engineering know-how, bugfixing, docker

**Reply by falkoschindler** [reply]
Working with Rodja at [Zauberzeug GmbH](https://zauberzeug.com/), I have similar background, motivation, expectations. Having a stable JustPy core that we can augment with NiceGUI is my main objective.

**Comment by rodja** [comment]
We also wrote a small motor controller UI called [odrive-gui](https://github.com/mawson-rovers/odrive-gui) which has been forked by [Mawson Rovers](https://mawsonrovers.com/) to fit [their own needs](https://github.com/mawson-rovers/odrive-gui).

**Comment by falkoschindler** [comment]
How unfortunate, but I didn't notice until now: All but the Berlin time are two hours off. So there might be folks from UK or US that missed our first meeting. 😕
